exercises/error_handling/errors4.rs

```diff
@@ -16,8 +14,7 @@ enum CreationError {
 
 impl PositiveNonzeroInteger {
     fn new(value: i64) -> Result<PositiveNonzeroInteger, CreationError> {
-        // Hmm... Why is this always returning an Ok value?
-        Ok(PositiveNonzeroInteger(value as u64))
+        if value == 0 { Err(CreationError::Zero) } else if value < 0 { Err(CreationError::Negative) } else { Ok(PositiveNonzeroInteger(value as u64)) }
```

Comment:
Please use [`u64::try_from<i64>`](https://doc.rust-lang.org/std/primitive.u64.html#impl-TryFrom%3Ci64%3E-for-u64) and use the appropriate error handling methods to map the output of `try_from` to the required error `enum`.

Also, what is the difference between `try_from` and `as` keyword?

Reply:
"as" performs an unchecked conversion, which means it will not throw an error if the value is not suitable for the target data type. This can lead to data loss, including issues like overflow and data truncation
"try_form" returns a Result object when the conversion is not feasible, making it a safer and more recommended approach.

Reply:
Great, notice that `as` should be avoided, just wanted to make sure we're aligned on this :)